python/ray/tests/test_tracing_task.py

```diff
@@ -0,0 +1,82 @@
+import os
+
+# TODO: use py_test(env = ...) in the build file with bazel 4.0
+os.environ["RAY_TRACING_ENABLED"] = "True"
+
+import shutil  # noqa: E402
+from opentelemetry import trace  # noqa: E402
+from opentelemetry.sdk.trace import TracerProvider  # noqa: E402
+from opentelemetry.sdk.trace.export import (
+    ConsoleSpanExporter,
+    SimpleExportSpanProcessor,
+)  # noqa: E402
+from typing import Any  # noqa: E402
+import ray  # noqa: E402
+
+
+spans_dir = "/tmp/spans"
+
+
+def _setup_tracing(*args: Any, **kwargs: Any) -> None:
+    if getattr(ray, "__traced__", False):
+        return
+
+    ray.__traced__ = True
+    # Sets the tracer_provider. This is only allowed once per execution
+    # context and will log a warning if attempted multiple times.
+    trace.set_tracer_provider(TracerProvider())
+    trace.get_tracer_provider().add_span_processor(
+        SimpleExportSpanProcessor(
+            ConsoleSpanExporter(
+                out=open(f"{spans_dir}/{os.getpid()}.txt", "w"),
+                formatter=lambda span: span.to_json(indent=None) + os.linesep,
+            )
+        )
+    )
+
+
+def test_tracing_task():
+    if os.path.exists(spans_dir):
+        shutil.rmtree(spans_dir)
+    os.mkdir(spans_dir)
```

Comment:
turn it into a fixture, so the code is cleaner https://docs.pytest.org/en/stable/fixture.html